---
title: "CSL-Portfolio_TW"
author: "Thijs (Mattheus) Wildschut"
date: "`r Sys.Date()`"
output: html_document
---

Built with R version `r getRversion()`

```{r load packages & functions, include=FALSE}
setwd("C:/Users/mwildschut/OneDrive - Vifor Pharma AG/Documents/R/Projects/R_CSL-Vifor_TW/CSL_Global-RPMC")
source("../SourceFile_TW.R")
HGNC = data.table::fread("../Other/Input/HGNC_GeneNames_17-05-23.txt", quote = "")

```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

# CSL assets with potential application in nephrology  {.tabset}

## OpenTargets overviews

### OpenTargets query function
```{r OpenTargets query function, include=FALSE}
library(httr)
HGNC = data.table::fread("../Other/Input/HGNC_GeneNames_04-05-23.csv")

gene_id = "PLG"
OT_query = function(gene_id){
  ensembl_id = na.omit(HGNC$`Ensembl ID(supplied by Ensembl)`[HGNC$`Approved symbol` == gene_id])[1]
  query_string = "
  query target($ensemblId: String!){
    target(ensemblId: $ensemblId){
      id
      approvedSymbol
      biotype
      associatedDiseases(page: { index: 0, size: 10000 }){
        count
        rows{
          disease{
            name
            therapeuticAreas{
              name
            }
            parents{
              name
              parents{
                name
                parents{
                  name
                  parents{
                    name
                    parents{
                      name
                      parents{
                        name
                        parents{
                          name
                          parents{
                            name
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
          score
          datatypeScores{
            id
            score
          }
        }
      }
    }
  }
"
  
  base_url <- "https://api.platform.opentargets.org/api/v4/graphql"
  variables <- list("ensemblId" = ensembl_id)
  post_body <- list(query = query_string, variables = variables)
  r <- POST(url=base_url, body=post_body, encode='json')
  
  data = httr::content(r)$data
  data3 = map_dfr(data$target$associatedDiseases$rows, function(x){
    c(setNames(c(x[[1]]$name, paste(unlist(x[[1]]$therapeuticAreas), collapse = " | "), paste(unlist(x[[1]]$parents), collapse = " | "), x[[2]]), 
               c("Disease_name", "Therapeutic_area", "Disease_parents", "Disease_score")), 
      x[[3]] %>% map_dfr(unlist) %>% pivot_wider(names_from = "id", values_from = "score"))
  }) %>% mutate(Kidney = factor(replace_na(str_extract(Disease_parents, "kidney disease|renal system measurement"), "Other"),
                                levels = c("kidney disease", "renal system measurement", "Other"), labels = c("Kidney disease", "Renal measurement", "Other"))) %>%
    group_by(Kidney) %>% slice_max(Disease_score, n = 25) %>% ungroup
  
  diseaseTypes = c("Disease_score", "genetic_association", "somatic_mutation", "known_drug", "affected_pathway", "literature", "rna_expression", "animal_model")
  diseaseTypeLabels = c("Overall\nassociation score", "Genetic associations", "Somatic mutations", "Drugs", "Pathways & systems biology",
                        "Text mining", "RNA expression", "Animal models")
  
  data4 = data3 %>%
    dplyr::mutate(across(.cols = !Disease_name & !Therapeutic_area & !Disease_parents & !Kidney, .fns = as.numeric)) %>%
    pivot_longer(cols = -Disease_name & -Therapeutic_area & -Disease_parents & -Kidney, names_to = "Scores", values_to = "Score") %>%
    mutate(Scores = factor(Scores, levels = diseaseTypes, labels = diseaseTypeLabels))
  
  areas = unique(as.vector(str_split(data3$Therapeutic_area, " \\| ", simplify = TRUE)))
  areas2 = map_dfc(as.list(areas[areas != ""]), function(x) setNames(data.frame(str_detect(data3$Therapeutic_area, x)), x)) %>%
    mutate("Disease_name" = data3$Disease_name) %>%
    pivot_longer(cols = -Disease_name) %>%
    `colnames<-`(c("Disease_name", "Area", "InArea"))

  data5 = data4 %>% left_join(areas2, by = "Disease_name", multiple = "all") %>%
    mutate(Disease_name = factor(Disease_name, levels = data3$Disease_name)) %>%
    complete(Scores, nesting(Disease_name, Therapeutic_area, Area, Kidney, InArea))

  plot1 = ggplot(data5, aes(x = Scores, y = Disease_name, fill = Score)) +
    geom_tile(col = "grey90") +
    scale_fill_viridis(na.value = "white") + scale_y_discrete(limits = rev) + scale_x_discrete(drop = FALSE) +
    facet_grid(rows = vars(Kidney), scales = "free", space = "free") +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), plot.title = element_text(hjust = 0.5), 
          strip.background = element_blank(), strip.text.y = element_blank(),
          text = element_text(size = 18)) + 
    labs(x = NULL, y = NULL, title = paste("Association scores", gene_id))
  
  plot2 = ggplot(data5, aes(x = Area, y = Disease_name, fill = InArea)) +
    geom_tile(col = "grey90") +
    scale_fill_manual(values = c("FALSE" = "white", "TRUE" = "black")) + scale_y_discrete(limits = rev) + 
    facet_grid(rows = vars(Kidney), scales = "free", space = "free") +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), plot.title = element_text(hjust = 0.5),
          strip.text.y = element_text(angle = 0), axis.text.y = element_blank(), axis.ticks.y = element_blank(),
          text = element_text(size = 18)) + 
    labs(x = NULL, y = NULL, title = "Disease areas") + guides(fill = "none")
  
  plots = plot1 + plot2 + plot_layout(guides = "collect", widths = c(1, 1.5))
  return(list("plots" = plots, "data4" = data4, "data5" = data5))
}

```

### Combined plot
```{r OpenTargets combined plot, fig.height=9, fig.width=12}
# gene.list = list("CR1", "C1QA", "C1QB", "C1QC", "C1R", "C1S", "C2", "C3", "C4A", "C4B", "C5")
gene.list = list("CCL17", "IL3RA", "LY96", "CXCR3", "PLG")

OT_list = map_dfr(gene.list, function(gene){
  data.frame("Gene" = gene, OT_query(gene)$data4)
})
data.OT = OT_list %>%
  filter(Kidney %in% c("Kidney disease", "Renal measurement")) %>%
  mutate(Scores = factor(Scores, levels = levels(Scores), 
                         labels = str_replace(str_replace_all(levels(Scores), " ", "\n"), "&\n", "& "))) %>%
  complete(Gene, Scores, nesting(Disease_name, Kidney))
dis.order = data.OT %>% filter(Scores == "Overall\nassociation\nscore" & !is.na(Score)) %>%
                     arrange(Gene, -Score) %>% select(Disease_name) %>% unique
data.OT = data.OT %>% mutate(Disease_name = factor(Disease_name, levels = dis.order$Disease_name))
ggplot(data.OT, aes(x = Gene, y = Disease_name, fill = Score)) +
  geom_tile(col = "grey90") +
  scale_fill_viridis(na.value = "white") + scale_y_discrete(limits = rev) + scale_x_discrete(drop = FALSE) +
  facet_grid(Kidney ~ Scores, scales = "free", space = "free") +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1), 
        strip.text.y = element_text(angle = 0), strip.text.x = element_text(size = 8))

ggsave("Output/OTquery_CSLassets-Combined_TW.pdf", height = 9, width = 12)

```

### Single plots
```{r OpenTargets single plots, fig.height=9, fig.width=18}
gene.list = list("CCL17", "IL3RA", "LY96", "CXCR3", "PLG")

plot.list = map(gene.list, function(gene){
  data.OT = OT_query(gene)$data4 %>%
    filter(Kidney %in% c("Kidney disease", "Renal measurement")) %>%
    complete(Scores, nesting(Disease_name, Kidney))
  dis.order = data.OT %>% filter(Scores == "Overall\nassociation score" & !is.na(Score)) %>%
    arrange(Score) %>% select(Disease_name) %>% unique
  data.OT = data.OT %>% 
    mutate(Disease_name = factor(Disease_name, levels = dis.order$Disease_name),
           Overall = factor(Scores == "Overall\nassociation score", levels = c(TRUE, FALSE)))
  ggplot(data.OT, aes(x = Scores, y = Disease_name, fill = Score)) +
    geom_tile(col = "grey90") +
    # scale_x_discrete(drop = FALSE) +
    facet_grid(Kidney ~ Overall, scales = "free", space = "free") +
    labs(x = NULL, y = NULL, title = gene) +
    theme_bw() + 
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1), 
          strip.text.y = element_text(angle = 0), plot.title = element_text(hjust = 0.5),
          strip.background.x = element_blank(), strip.text.x = element_blank())
})
max.score = max(unlist(map(plot.list, function(plot) plot$data$Score)), na.rm = TRUE)
plot.list = map(plot.list, function(plot) plot + scale_fill_viridis(na.value = "white", limits = c(0, max.score)))
wrap_plots(plot.list, guides = "collect", heights = c(1,2))

ggsave("Output/OTquery_CSLassets-Singles_TW.pdf", height = 9, width = 18)

```


## AKI

### Urine CSA-AKI
Fetuin-A = AHSG = P02765 = FETUA_Human = Alpha-2-HS-glycoprotein

- Paper URL: https://www.frontiersin.org/articles/10.3389/fbioe.2022.1002853/full
- Data downloaded from: https://www.frontiersin.org/articles/10.3389/fbioe.2022.1002853/full#supplementary-material
  - Table 4.xls
- Gene names downloaded from: https://www.genenames.org/download/custom/
  - Selected columns: all standards + "Alias names", "NCBI Gene ID", "Ensembl gene ID", "UniProt ID"
  - Saved as .txt

```{r Fetuin-A, fig.height=4, fig.width=7}
raw.data = fread("Input/Bai2022_UrinaryProteomics_Table 4.csv")
HGNC = data.table::fread("../Other/Input/HGNC_GeneNames_17-05-23.txt", quote = "")

data = raw.data %>%
  select(-ProteinGroup) %>%
  mutate(Protein = HGNC$`Approved symbol`[match(str_extract(Protein, "(?<=sp\\|).*(?=\\|)"),
                                                 HGNC$`UniProt ID(supplied by UniProt)`)]) %>%
  filter(!is.na(Protein)) %>% # 16/1942 Proteins not matched, 1926 remaining
  column_to_rownames("Protein") %>%
  t %>% as.data.frame %>%
  mutate(Patient = factor(rep(1:6, 2)),
         TimePoint = factor(rep(c("After surgery (Day 1)", "Before surgery"), each = 6),
                            levels = c("Before surgery", "After surgery (Day 1)")))

data = data %>% mutate(norm_FetuinA = AHSG - CST3)

gene.list = list("CCL17", "IL3RA", "LY96", "CXCR3", "PLG")
plot.list = map(gene.list[gene.list %in% colnames(data)], function(gene){
  box.plot = ggplot(data, aes(x = TimePoint, y = .data[[gene]], fill = TimePoint, col = TimePoint)) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0) + geom_jitter(alpha = 0.25, width = 0.2) +
    stat_compare_means(comparisons = list(c(1,2)), size = 3) +
    theme_bw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + xlab(NULL)
  
  line.plot = ggplot(data, aes(x = TimePoint, y = .data[[gene]], group = Patient, col = Patient)) +
    geom_line() + geom_point() +
    theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
  
  box.plot #+ line.plot
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("CSA-AKI mouse model urinary proteomics\nBai Front. Bioeng. Biotechnol. 2022") & theme(plot.title = element_text(hjust = 0.5))

# ggsave("Output/Bai2022_UrinaryProteomics_FetuinA.pdf", height = 4, width = 7)

```

### Mouse model of septic AKI

- Paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7710145/
- Data downloaded: Supplementary Table S1 ("162338_1_supp_596735_qgmqcy.xlsx")

```{r Septic-AKI mouse, fig.width=4, fig.height=4}
raw.data = read.xlsx("Input/AKI/162338_1_supp_596735_qgmqcy.xlsx", sheetIndex = 1, startRow = 2)
data = raw.data %>% dplyr::select(Gene.Name, contains("LFQ")) %>%
  column_to_rownames("Gene.Name") %>% t %>% as.data.frame %>%
  # dplyr::mutate(across(everything(), ~log2(.x+1))) %>%
  mutate(Animal = as.numeric(str_extract(rownames(.), "(?<=G)[:digit:]{1,2}"))) %>%
  arrange(Animal) %>%
  mutate(Group = rep(c("Day 0", "Day 2", "Day 7"), each = 6))

gene = "Il6r"
ggplot(data, aes(x = Group, y = .data[[gene]], col = Group, fill = Group)) +
  geom_boxplot(alpha = 0.25, outlier.alpha = 0) + geom_jitter(width = 0.2, alpha = 0.2) +
  stat_compare_means(comparisons = list(c(1,2), c(1,3), c(2,3)), size = 4) +
  labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), 
       title = "Kidney proteomics\nSepsis-induced Kidney Injury mouse model\nLin | Mol Cell Proteomics 2020") +
  theme_bw() + 
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.title = element_text(hjust = 0.5, size = 12))

sort(colnames(data)[str_detect(colnames(data), "Il|il")])

c("Ccl17", "Ly96", "Plg", "Il3ra", "Cxcr3")[c("Ccl17", "Ly96", "Plg", "Il3ra", "Cxcr3") %in% colnames(data)]

```


## Lupus Nephritis
Highlighted drugs:

- CSL305 - anti-**C2** mAb - dual FcRn blocker & complement CP & LP inhibitor
- CSL040 - soluble **CR1** - complement pathway inhibitor
- CSL300 - anti-**IL-6** - anti-inflammatory
- Vamifeport - Ferroportin/**SLC40A1**

Strategy taken: scRNAseq data of LN patients

Data downloaded from: https://singlecell.broadinstitute.org/single_cell/study/SCP279/amp-phase-1

- Paper 1: AMP Phase 1 - Dataset 2: Lupus Nephritis (Broad Institute) 
  - https://www.nature.com/articles/s41590-019-0398-x
  - Arazi *et al.*, The immune cell landscape in kidneys of patients with lupus nephritis, *Nature Immunology* 2019
- Paper 2: AMP Phase 1 - Dataset 3: Lupus Nephritis (NYU Langone - Metro)
  - https://www.nature.com/articles/s41590-019-0386-1
  - Der *et al.*, Tubular cell and keratinocyte single-cell transcriptomics applied to lupus nephritis reveal type I IFN and fibrosis relevant pathways, **Nature Immunology**, 2019

### AMP single-cell data
```{r Load data}
annot = fread("Input/LupusNephritis/meta.gz")
data1 = fread("Input/LupusNephritis/exprMatrixSleBroad.tsv.gz")
data2 = fread("Input/LupusNephritis/exprMatrixSleMetro.tsv.gz")


```

<!-- ### Set up query functions -->
<!-- ```{r Query functions} -->

<!-- ``` -->

### Bancheraeu - longitudinal SLE

- Paper: Banchereau et al., "Personalized Immunomonitoring Uncovers Molecular Networks that Stratify Lupus Patients", Cell 2016
- Link paper: https://www.sciencedirect.com/science/article/pii/S0092867416302641
```{r Lupus bulk transcriptomics, fig.width=4.25, fig.height=3.5}
data3 = getGEO("GSE65391")[[1]]
annot.pat = data3@phenoData@data %>%
  filter(`set:ch1` != "Technical_Replicate") %>%
  `colnames<-`(str_replace(str_remove(colnames(.), "\\:ch1"), " ", "_")) %>%
  dplyr::select(geo_accession, subject, disease_state, visit, disease_activity, sledai, treatment_lmm1, sledaic_lmm2) %>%
  mutate(sledai_grouped = factor(sledaic_lmm2, levels = sort(unique(sledaic_lmm2)), 
                                 labels = rep(c("Renal", "No renal", "Healthy"), c(2,3,1)))) %>%
  mutate(sledai_grouped = factor(sledai_grouped, levels = c("Healthy", "No renal", "Renal"),
                                 labels = c("Healthy (48)", "SLE | No renal (516)", "SLE | Renal (408)")))

annot.gene = data3@featureData@data
expr = data3@assayData$exprs %>% as.data.frame %>%
  dplyr::mutate(Gene = annot.gene$ILMN_Gene[match(rownames(.), annot.gene$ID)]) %>%
  group_by(Gene) %>% dplyr::summarise(across(everything(), max))
expr2 = expr %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% inner_join(annot.pat, by = "geo_accession") %>%
  mutate(disease_activity = factor(ifelse(disease_state == "Healthy", "Healthy", paste0("SLE_", disease_activity)),
                                   levels = c("Healthy", paste0("SLE_", 1:3))))

pat.1 = expr2 %>% group_by(disease_activity) %>% dplyr::count() %>%
  mutate(Labels = paste0(disease_activity, " (", n, ")"))
pat.2 = expr2 %>% group_by(sledaic_lmm2) %>% dplyr::count() %>%
  mutate(Labels = paste0(sledaic_lmm2, " (", n, ")"))
expr2 = expr2 %>% mutate(disease_activity = factor(disease_activity, 
                                                   levels = pat.1$disease_activity, labels = pat.1$Labels),
                         sledaic_lmm2 = factor(sledaic_lmm2,
                                               levels = rev(pat.2$sledaic_lmm2), labels = rev(pat.2$Labels)))

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")#, "IFIT3", "CXCL9", "CXCL10", "CXCL11")
gene.list = list("LY96")#, "CXCR3")#, "CXCL10", "IFIT3")
# gene.list = list("IL6R", "IL6", "IL5", "IL4")
var = "sledai_grouped"# "disease_activity" # #"sledaic_lmm2" # 
plot.list = map2(gene.list[gene.list %in% colnames(expr2)], var, function(gene, score){
  grouping = ifelse(score == "disease_activity", "Disease activity", "SLEDAI categories")
  comp.list = if(score == "disease_activity") list(c(1,2), c(2,3), c(3,4), c(1,3), c(1,4)) else list(c(1,2), c(1,3), c(2,3))#list(c(1,2), c(2,3), c(3,4), c(4,5), c(5,6), c(3,5), c(2,5))
  ggplot(expr2, aes(x = .data[[score]], y = .data[[gene]], fill = .data[[score]], col = .data[[score]])) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.15) +
    stat_compare_means(comparisons = comp.list, na.rm = TRUE, size = 3, step.increase = 0.075) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = grouping, col = grouping) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("SLE longitudinal whole blood transcriptomics\nBanchereau Cell 2016") & theme(plot.title = element_text(hjust = 0.5))
ggsave("Output/SLE_BloodTranscriptomics_LY96_DA.pdf", width = 4.25, height = 3.5)
# ggsave("Output/SLE_BloodTranscriptomics_CSL-assets_DA.pdf", width = 10, height = 7)

```

### Berthier Lupus Glom & TubInt

- Paper: Berthier et al. (Kretzler group), "Cross-Species Transcriptional Network Analysis Defines Shared Inflammatory Responses in Murine and Human Lupus Nephritis", The Journal of Immunology 2012
- Link: https://journals.aai.org/jimmunol/article/189/2/988/86778/Cross-Species-Transcriptional-Network-Analysis
```{r Berthier Lupus Glom, fig.width=5, fig.height=3.25}
data = getGEO("GSE32591")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  select(geo_accession, disease_status, patient_number, tissue) %>%
  mutate(Tissue = str_remove(tissue, " from kidney biopsy"))
pat.counts = annot.pat %>% group_by(disease_status) %>% dplyr::count(Tissue) %>% ungroup %>%
  mutate(Labels = paste0(capitalize(disease_status), "_", Tissue, " (", n, ")"))
annot.pat = annot.pat %>% left_join(pat.counts, by = c("disease_status", "Tissue")) %>%
  mutate(disease_status = factor(capitalize(disease_status), levels = c("Control", "LN patient"),
                                 labels = c("Control (n=14/15)", "LN Patient (n=32)")))
annot.gene = data@featureData@data
expr = data@assayData$exprs %>% as.data.frame %>%
  mutate(Gene = annot.gene$`Gene Symbol`[match(rownames(.), annot.gene$ID)]) %>%
  group_by(Gene) %>% dplyr::summarise(across(everything(), max)) %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
gene.list = list("LY96")
plot.list = map(gene.list[gene.list %in% colnames(expr)], function(gene){
  ggplot(expr, aes(x = disease_status, y = .data[[gene]], fill = disease_status, col = disease_status)) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    facet_wrap(.~Tissue) +
    stat_compare_means(comparisons = list(c(1,2), c(3,4)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("LN transcriptomics\nBerthier J Immunol 2012") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/LN_GlomTubInt-Transcriptomics_All.pdf", width = 7.7, height = 5.25)
ggsave("Output/LN_GlomTubInt-Transcriptomics_LY96.pdf", width = 5, height = 3.25)

```
### LN Mejia-Vilet flare transcriptomics

- Paper: Mejia-Vilet et al., "Immune gene expression in kidney biopsies of lupus nephritis patients at diagnosis and at renal flare", Nephrology Dialysis Transplantation 2019
- Link: https://academic.oup.com/ndt/article/34/7/1197/5004399
```{r LN Mejia-Vilet flare transcriptomics, fig.width=7, fig.height=3.25}
data = getGEO("GSE113342")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace_all(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Tissue = str_extract(tissue, "glomeruli|tubulointerstitium"),
         Patient = str_extract(title, "Patient [:digit:]{1,2}|control [:digit:]{1,2}"),
         Disease = ifelse(lupus_nephritis_class == "normal kidney tissue", "Control", "LN Patient"),
         Visit = replace_na(str_extract(source_name_ch1, "first|repeat"), "Control")) %>%
  select(geo_accession, Disease, lupus_nephritis_class, Patient, Tissue, Visit)
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count(Tissue) %>% ungroup %>%
  mutate(Labels = paste0(capitalize(Disease), "_", Tissue, " (", n, ")"))
# annot.pat = annot.pat %>% left_join(pat.counts, by = c("disease_status", "Tissue")) %>%
#   mutate(disease_status = factor(capitalize(disease_status), levels = c("Control", "LN patient"),
#                                  labels = c("Control (n=14/15)", "LN Patient (n=32)")))
annot.gene = data@featureData@data
expr = data@assayData$exprs %>%
  t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
gene.list = list("LY96")
plot.list = map(gene.list[gene.list %in% colnames(expr)], function(gene){
  ggplot(expr, aes(x = Visit, y = .data[[gene]], fill = Visit, col = Visit)) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    facet_wrap(.~Tissue) +
    # stat_compare_means(comparisons = list(c(1,2), c(3,4)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("LN transcriptomics\nMejia-Vilet NDT 2019") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/LN_GlomTubInt-Transcriptomics_All.pdf", width = 7.7, height = 5.25)
# ggsave("Output/LN_GlomTubInt-Transcriptomics_LY96.pdf", width = 5, height = 3.25)

```

### LN Wither

- Link https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0196117

```{r LN Wither, fig.width=4.25, fig.height=3.25}
data = getGEO("GSE99967")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Disease = replace_na(nephritis_state, "Control")) %>%
  select(geo_accession, Disease)
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Labels = factor(paste0(str_remove(capitalize(Disease), " \\(.*\\)"), " (", n, ")"),
                         levels = c("Control (17)", "Without LN (13)", "Active LN (29)"),
                         labels = c("Control (17)", "SLE | Without LN (13)", "SLE | Active LN (29)")))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")
annot.gene = data@featureData@data
expr = data@assayData$exprs %>% as.data.frame %>%
  mutate(Gene = annot.gene$GeneSymbol[match(rownames(.), annot.gene$ID)]) %>%
  group_by(Gene) %>% dplyr::summarise(across(everything(), max)) %>%
  filter(!is.na(Gene)) %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
gene.list = list("LY96")
var = "Labels"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    stat_compare_means(comparisons = list(c(1,2), c(2,3), c(1,3)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("SLE Whole Blood transcriptomics\nWither PlosOne 2018") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/LN_GlomTubInt-Transcriptomics_All.pdf", width = 7.7, height = 5.25)
ggsave("Output/SLE-LN-Wither-Transcriptomics_LY96.pdf", width = 4.25, height = 3.5)

```
### Chiche blood longitudinal SLE

```{r Chiche blood longitudinal SLE, fig.width=4.25, fig.height=3.25}
data = getGEO("GSE49454")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Disease = replace_na(renal, "Control")) %>%
  select(geo_accession, Disease)
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Labels = factor(paste0(str_remove(capitalize(Disease), " \\(.*\\)"), " (", n, ")"),
                         levels = c("Control (20)", "N (93)", "Y (64)"),
                         labels = c("Control (20)", "SLE | Without LN (93)", "SLE | Active LN (64)")))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")
annot.gene = data@featureData@data
expr = data@assayData$exprs %>% as.data.frame %>%
  mutate(Gene = annot.gene$Symbol[match(rownames(.), annot.gene$ID)]) %>%
  filter(!is.na(Gene)) %>%
  group_by(Gene) %>% dplyr::summarise(across(everything(), max)) %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
gene.list = list("LY96")
var = "Labels"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    stat_compare_means(comparisons = list(c(1,2), c(2,3), c(1,3)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("SLE Whole Blood transcriptomics\nChiche Arthritis Rhematol. 2014") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/LN_GlomTubInt-Transcriptomics_All.pdf", width = 7.7, height = 5.25)
ggsave("Output/SLE-LN-Cliche-Transcriptomics_LY96.pdf", width = 4.25, height = 3.5)
```

## CKD

### CKD plasma proteomics

- Paper: Medina et al., Proteomic signature associated with chronic kidney disease (CKD) progression identified by data-independent acquisition mass spectrometry, Clinical Proteomics 2023
- Data downloaded from: https://clinicalproteomicsjournal.biomedcentral.com/articles/10.1186/s12014-023-09405-0
  - Additional file 2

```{r CKD plasma proteomics, fig.width=3.75, fig.height=3.25}
data = read.csv("Input/12014_2023_9405_MOESM2_ESM.csv")
pat.counts = data %>% group_by(Progressor.type) %>% dplyr::count() %>% 
  ungroup %>% dplyr::select(n) %>% unlist
data = data %>%
  mutate(Progressor.type = factor(Progressor.type, levels = c("Stable", "Rapid"),
                                  labels = paste0(c("Stable", "Rapid"), " (", rev(pat.counts), ")")))

colnames(data)[-c(1,ncol(data))] = HGNC$`Approved symbol`[match(colnames(data)[-c(1,ncol(data))],
                                                                HGNC$`UniProt ID(supplied by UniProt)`)]
  
ggplot(data, aes(x = Progressor.type, y = PLG, col = Progressor.type, fill = Progressor.type)) +
  geom_boxplot(alpha = 0.25, outlier.alpha = 0) + geom_jitter(alpha = 0.2, width = 0.2) +
  stat_compare_means(comparisons = list(c(1,2)), size = 3) +
  labs(x = NULL, y = "PLG - Plasmin(ogen)\nProtein levels (log2)", title = "CKD patient plasma proteomics\nMedina | Clinical Proteomics 2023") +
  theme_bw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.title = element_text(hjust = 0.5))
ggsave("Output/CKD-plasma-proteomics_PLG.pdf", width = 3.75, height = 3.25)

```

## IgAN

### Cox IgAN Blood 2

- Paper: Cox et al., "Altered modulation of WNT-beta-catenin and PI3K/Akt pathways in IgA nephropathy", Kidney Int. 2010
- Link: https://www.sciencedirect.com/science/article/pii/S0085253815545456
```{r NephroSeq Cox IgAN Blood 2, fig.width=6, fig.height=4.5}
data = getGEO("GSE14795")[[2]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Disease = str_extract(title, "Healthy|IGAN")) %>%
  select(geo_accession, Disease)
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Labels = paste0(Disease, " (", n, ")"))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")

annot.gene = data@featureData@data
expr = data@assayData$exprs %>% as.data.frame %>%
  mutate(Gene = annot.gene$`Gene Symbol`[match(rownames(.), annot.gene$ID)]) %>%
  group_by(Gene) %>% dplyr::summarise(across(everything(), ~max(log2(.x)))) %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
# gene.list = list("IL6", "IL6R")
# gene.list = list("MVD", "HSP90AB1", "JAK2")
var = "Disease"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    stat_compare_means(comparisons = list(c(1,2), c(3,4)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("IgAN Whole blood transcriptomics\nCox Kid Int. 2010") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/IgAN_CoxBlood2_All.pdf", width = 6, height = 4.5)

```

### Cox IgAN Monocytes

- Paper: Cox et al., "Altered monocyte expression and expansion of non-classical monocyte subset in IgA nephropathy patients", Nephrol Dial Transplant 2015
- Link: https://www.sciencedirect.com/science/article/pii/S0085253815545456
```{r Cox IgAN Monocytes, fig.width=8.5, fig.height=4.5}
data = getGEO("GSE58539")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Disease = str_extract(disease_status, "Healthy|IgAN")) %>%
  select(geo_accession, Disease)
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Labels = paste0(Disease, " (", n, ")"))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")

annot.gene = data@featureData@data
expr = data@assayData$exprs %>% as.data.frame %>%
  dplyr::mutate(Gene = annot.gene$Symbol[match(rownames(.), annot.gene$ID)]) %>%
  group_by(Gene) %>% dplyr::summarise(across(everything(), ~max(.x))) %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
gene.list = list("IL6", "IL6R")
# gene.list = list("MVD", "HSP90AB1", "JAK2")
var = "Labels"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    stat_compare_means(comparisons = list(c(1,2), c(3,4)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("IgAN CD14+ Monocyte transcriptomics\nCox Nephrol Dial Transplant 2015") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/IgAN_CoxBlood2_All.pdf", width = 6, height = 4.5)

```


## FSGS (Nephrotic Syndrome)

### Mariani Nephrotic Syndrome TI

- Paper: Mariani et al. (Kretzler group), "Precision nephrology identified tumor necrosis factor activation variability in minimal change disease and focal segmental glomerulosclerosis", Kidney International 2023
- Link: https://www.sciencedirect.com/science/article/pii/S0085253822009668
- Data downloaded from: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE219185
```{r Mariani Nephrotic Syndrome TI, fig.width=8.5, fig.height=4.5}
raw.data = fread("Input/GSE219185_TI_NEPTUNE.txt.gz") %>%
  dplyr::select(contains(c("gene", "Sample")))
data = getGEO("GSE219185")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Sample = str_extract(title, "Sample_[:digit:]{1,3}"),
         Disease = disease) %>%
  select(geo_accession, Disease, Sample)
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Labels = factor(paste0(Disease, " (", n, ")"), levels = rev(paste0(Disease, " (", n, ")"))))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")

HGNC = data.table::fread("../Other/Input/HGNC_GeneNames_17-05-23.txt", quote = "")
expr = raw.data %>%
  mutate(Gene = HGNC$`Approved symbol`[match(Ensembl_gene_ID, HGNC$`Ensembl gene ID`)]) %>%
  dplyr::select(-Ensembl_gene_ID) %>%
  filter(!is.na(Gene)) %>%
  # group_by(Gene) %>% dplyr::summarise(across(everything(), max)) %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("Sample") %>% left_join(annot.pat, by = "Sample")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
# gene.list = list("MVD", "HSP90AB1", "JAK2")
gene.list = list("IL6", "IL6R")
var = "Labels"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    stat_compare_means(comparisons = list(c(1,2), c(3,4)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("FSGS transcriptomics\nMariani et al Kidney International 2023") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/IgAN_CoxBlood2_All.pdf", width = 6, height = 4.5)

```

## Across diseases

### Neptune Glomerular RNAseq

- Paper: Sealfon et al. (Kretzler/Trayanakaya/Beck group), "Molecular Characterization of Membranous Nephropathy", JASN 2022
- Link: https://journals.lww.com/jasn/Abstract/2022/06000/Molecular_Characterization_of_Membranous.18.aspx
- Data downloaded from: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE197307
```{r NEPTUNE Glom, fig.width=10, fig.height=4.5}
raw.data = fread("Input/GSE197307_NEPTUNE_Glom_RNA_counts_2022.txt.gz")
data = getGEO("GSE197307")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Sample = title,
         Disease = disease) %>%
  select(geo_accession, Disease, Sample)
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Labels = factor(paste0(Disease, " (", n, ")"),
                         levels = c("Healthy living transplant donor (8)", "Minimal Change Disease (89)",
                                    "Focal Segmental Glomerulosclerosis (93)", "Membranous Nephropathy (62)",
                                    "Other Nephrotic Syndrome (29)", "Nephrotic Syndrome, unspecified (1)")))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")

HGNC = data.table::fread("../Other/Input/HGNC_GeneNames_17-05-23.txt", quote = "")
expr = raw.data %>%
  mutate(Gene = HGNC$`Approved symbol`[match(str_replace(ENSEMBL_ene_ID, "ENS", "ENSG"), HGNC$`Ensembl gene ID`)]) %>%
  dplyr::select(-ENSEMBL_ene_ID) %>%
  filter(!is.na(Gene)) %>%
  # group_by(Gene) %>% dplyr::summarise(across(everything(), max)) %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  dplyr::mutate(across(everything(), ~log2(.x + 1))) %>%
  rownames_to_column("Sample") %>% #dplyr::mutate(Sample = str_extract(Sample, "Sample_[:digit:]{1,3}")) %>%
  left_join(annot.pat, by = "Sample")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
# gene.list = list("MVD", "HSP90AB1", "JAK2")
# gene.list = list("IL6", "IL6R")
var = "Labels"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_hline(yintercept = median(expr[[gene]][expr$Disease == "Healthy living transplant donor"]), linetype = 2) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    # stat_compare_means(comparisons = list(c(1,2), c(1,3), c(1,4), c(1,5), c(2,3), c(2,4), c(2,5)),
                       # na.rm = TRUE, size = 3, label = "p.signif", vjust = 0.45, hide.ns = TRUE) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.05))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("Neptune Cohort\nGlomerular transcriptomics (RNAseq)") & theme(plot.title = element_text(hjust = 0.5))
ggsave("Output/Neptune-Glomerular_All.pdf", width = 10, height = 4.5)

```

### Neptune TI Adult RNAseq
- Paper: 
- Link: 
- Data downloaded from: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE182380
```{r Neptune TI, fig.width=10, fig.height=4.5}
raw.data = fread("Input/GSE182380_NEPTUNE_TI_adult_RNA_counts_2021.txt.gz")
data = getGEO("GSE182380")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Sample = str_extract(title, "Sample_[:digit:]{1,3}"),
         Disease = disease) %>%
  select(geo_accession, Disease, Sample)
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Labels = factor(paste0(Disease, " (", n, ")"),
                         levels = c("Healthy living transplant donor (0)", "Minimal Change Disease (32)",
                                    "Focal Segmental Glomerulosclerosis (70)", "Membranous Nephropathy (59)",
                                    "Other Nephrotic Syndrome (37)", "Nephrotic Syndrome, unspecified (1)")))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")

HGNC = data.table::fread("../Other/Input/HGNC_GeneNames_17-05-23.txt", quote = "")
expr = raw.data %>%
  column_to_rownames("GeneSymbol") %>% t %>% as.data.frame %>%
  dplyr::mutate(across(everything(), ~log2(.x + 1))) %>%
  rownames_to_column("Sample") %>% #dplyr::mutate(Sample = str_extract(Sample, "Sample_[:digit:]{1,3}")) %>%
  left_join(annot.pat, by = "Sample")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
# gene.list = list("ACTB", "GAPDH")

# gene.list = list("MVD", "HSP90AB1", "JAK2")
var = "Labels"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_hline(yintercept = median(expr[[gene]][expr$Disease == "Minimal Change Disease"]), linetype = 2) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    # stat_compare_means(comparisons = list(c(2,3), c(2,4), c(2,5)),
                       # na.rm = TRUE, size = 3, label = "p.signif", vjust = 0.45, hide.ns = TRUE) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.05))) +
    scale_x_discrete(drop = FALSE) + scale_fill_discrete(drop = FALSE) + scale_color_discrete(drop = FALSE) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("Neptune Cohort\nTubulointerstitial transcriptomics (RNAseq)") & theme(plot.title = element_text(hjust = 0.5))
ggsave("Output/Neptune-Tubulointerstitial_All.pdf", width = 10, height = 4.5)

```

### Neptune TI Array
- Data from: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108113
- Alternative: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE133288

### ERCB Glomerular
- Data from: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE104948

```{r ERCB Glomerular, fig.width=7.5, fig.height=3.75}
data = getGEO("GSE104948")
annot.pat = rbind.data.frame(data[[1]]@phenoData@data, data[[2]]@phenoData@data) %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Sample = str_remove(title, ".*-"),
         Disease = replace_na(str_replace(diagnosis, "Membrande", "Membrane"), "Control")) %>%
  select(geo_accession, Disease, Sample)

disease.order = c("Control", "Tumor Nephrectomy", "Minimal Change Disease", "Focal Segmental Glomerular Sclerosis/Minimal Change Disease", "Focal Segmental Glomerular Sclerosis", "Thin Membrane Disease", "Membranous Glomerulonephropathy", "IgA Nephropathy", "Diabetic Nephropathy", "Hypertensive Nephropathy", "ANCA Associated Vasculitis", "Systemic Lupus Erythematosus")
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Disease = factor(Disease, levels = disease.order)) %>%
  arrange(Disease) %>%
  mutate(Labels = factor(Disease, labels = paste0(Disease, " (", n, ")")))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")

annot.gene = data[[1]]@featureData@data
expr = cbind.data.frame(data[[1]]@assayData$exprs, data[[2]]@assayData$exprs) %>%
  mutate(Gene = annot.gene$ENTREZ_GENE_ID[match(rownames(.), annot.gene$ID)]) %>%
  mutate(Gene = HGNC$`Approved symbol`[match(Gene, HGNC$`NCBI Gene ID`)]) %>%
  filter(!is.na(Gene)) %>%
  # group_by(Gene) %>% dplyr::summarise(across(everything(), max)) %>%
  remove_rownames %>% column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
gene.list = list("LY96")
var = "Labels"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_hline(yintercept = median(expr[[gene]][expr$Disease == "Control"]), linetype = 2) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    # stat_compare_means(comparisons = list(c(1,2), c(3,4)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("European Renal cDNA Bank\nGlomerular transcriptomics (array profiling)") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/ERCB-Glomerular_All.pdf", width = 12.5, height = 5)
ggsave("Output/ERCB-Glomerular_LY96.pdf", width = 7.5, height = 3.75)

```

### ERCB Tubulointerstitial
- Data from: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE104954

```{r ERCB TI, fig.width=7.5, fig.height=3.75}
data = getGEO("GSE104954")
annot.pat = rbind.data.frame(data[[1]]@phenoData@data, data[[2]]@phenoData@data) %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Sample = str_remove(title, ".*-"),
         Disease = replace_na(str_replace(diagnosis, "membrande", "membrane"), "Control")) %>%
  select(geo_accession, Disease, Sample)

disease.order = c("Control", "Tumor nephrectomy", "Minimal change disease", "Focal segmental glomerulosclerosis and minimal change disease", "Focal segmental glomerulosclerosis", "Thin membrane disease", "Membranous glomerulonephropathy", "IgA nephropathy", "Diabetic nephropathy", "Hypertensive nephropathy", "ANCA-associated vasculitis", "Systemic lupus erythematosus")
pat.counts = annot.pat %>% group_by(Disease) %>% dplyr::count() %>% ungroup %>%
  mutate(Disease = factor(Disease, levels = disease.order)) %>%
  arrange(Disease) %>%
  mutate(Labels = factor(Disease, labels = paste0(Disease, " (", n, ")")))
annot.pat = annot.pat %>% left_join(pat.counts, by = "Disease")

annot.gene = data[[1]]@featureData@data
expr = cbind.data.frame(data[[1]]@assayData$exprs, data[[2]]@assayData$exprs) %>%
  mutate(Gene = annot.gene$ENTREZ_GENE_ID[match(rownames(.), annot.gene$ID)]) %>%
  mutate(Gene = HGNC$`Approved symbol`[match(Gene, HGNC$`NCBI Gene ID`)]) %>%
  filter(!is.na(Gene)) %>%
  # group_by(Gene) %>% dplyr::summarise(across(everything(), max)) %>%
  remove_rownames %>% column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
gene.list = list("LY96")
var = "Labels"
plot.list = map2(gene.list[gene.list %in% colnames(expr)], var, function(gene, var){
  ggplot(expr, aes(x = .data[[var]], y = .data[[gene]], fill = .data[[var]], col = .data[[var]])) +
    geom_hline(yintercept = median(expr[[gene]][expr$Disease == "Control"]), linetype = 2) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    # facet_wrap(.~Tissue) +
    # stat_compare_means(comparisons = list(c(1,2), c(3,4)), na.rm = TRUE, size = 3) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = "Disease status", col = "Disease status") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("European Renal cDNA Bank\nTubulointerstitial transcriptomics (array profiling)") & theme(plot.title = element_text(hjust = 0.5))
# ggsave("Output/ERCB-Tubulointerstitial_All.pdf", width = 12.5, height = 5)
ggsave("Output/ERCB-Tubulointerstitial_LY96.pdf", width = 7.5, height = 3.75)

```

## Transplant

### Transplants Flechner

```{r Flechner transplant kidney&PBL, fig.width=9.5, fig.height=5.5}
data = getGEO("GSE1563")[[1]]
annot.pat = data@phenoData@data %>%
  `colnames<-`(str_replace(str_remove(colnames(.), ":ch1"), " ", "_")) %>%
  mutate(Tissue = factor(source_name_ch1, levels = c("kidney", "PBL"), labels = c("Kidney tissue", "Peripheral blood lymphocytes")),
         Condition = str_extract(description, 
                                 "healthy|acute rejection|well-functioning|renal dysfunction")) %>%
  mutate(Condition = factor(Condition, levels = c("healthy", "well-functioning", "renal dysfunction", "acute rejection"),
                            labels = c("Control | Healthy", "Transplant | Well-functioning", "Transplant | Renal dysfunction", "Transplant | Acute rejection"))) %>%
  select(geo_accession, Tissue, Condition) %>%
  arrange(Tissue, Condition)
pat.counts = annot.pat %>% group_by(Tissue, Condition) %>% dplyr::count() %>% ungroup %>%
  mutate(Labels = paste0(Condition, " (", n, ")")) %>%
  mutate(Labels = factor(Labels, levels = Labels))
annot.pat = annot.pat %>% left_join(pat.counts, by = c("Condition", "Tissue"))
annot.gene = data@featureData@data
expr = data@assayData$exprs %>% as.data.frame %>%
  mutate(Gene = annot.gene$`Gene Symbol`[match(rownames(.), annot.gene$ID)]) %>%
  filter(!is.na(Gene)) %>%
  group_by(Gene) %>% dplyr::summarise(across(everything(), max)) %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% left_join(annot.pat, by = "geo_accession")

gene.list = c("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")
# gene.list = list("LY96")
var = "Labels"
plot.list = map2(as.list(levels(expr$Tissue)), var, function(tissue, var){
  expr.long = expr %>%
    filter(Tissue == tissue) %>%
    dplyr::select(var, all_of(gene.list[gene.list %in% colnames(expr)])) %>%
    pivot_longer(cols = -var, names_to = "Gene", values_to = "Expression") %>%
    mutate(Expression = log2(Expression))
  ggplot(expr.long, aes(x = .data[[var]], y = Expression, fill = .data[[var]], col = .data[[var]])) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.25) +
    facet_wrap(.~Gene, scales = "free", nrow = 1) +
    stat_compare_means(comparisons = list(c(2,3), c(2,4), c(3,4)), na.rm = TRUE, size = 2.5) +
    labs(x = NULL, y = "mRNA levels (log2)", title = tissue, col = "Condition", fill = "Condition") +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.1))) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, nrow = 2) + 
  plot_annotation("Transplant transcriptomics\nFlechner Am J Transplant. 2004") & theme(plot.title = element_text(hjust = 0.5))
ggsave("Output/Transplant-Flechner-Transcriptomics_All.pdf", width = 9.5, height = 5.5)

```




## Other

