---
title: "CSL-Portfolio_TW"
author: "Thijs (Mattheus) Wildschut"
date: "`r Sys.Date()`"
output: html_document
---

Built with R version `r getRversion()`

```{r load packages & functions, include=FALSE}
setwd("C:/Users/mwildschut/OneDrive - Vifor Pharma AG/Documents/R/Projects/R_CSL-Vifor_TW/CSL_Global-RPMC")
source("../SourceFile_TW.R")

```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```

# CSL assets with potential application in nephrology

## OpenTargets overviews

### OpenTargets query function
```{r OpenTargets query function, include=FALSE}
library(httr)
HGNC = data.table::fread("../Other/Input/HGNC_GeneNames_04-05-23.csv")

gene_id = "PLG"
OT_query = function(gene_id){
  ensembl_id = na.omit(HGNC$`Ensembl ID(supplied by Ensembl)`[HGNC$`Approved symbol` == gene_id])[1]
  query_string = "
  query target($ensemblId: String!){
    target(ensemblId: $ensemblId){
      id
      approvedSymbol
      biotype
      associatedDiseases(page: { index: 0, size: 10000 }){
        count
        rows{
          disease{
            name
            therapeuticAreas{
              name
            }
            parents{
              name
              parents{
                name
                parents{
                  name
                  parents{
                    name
                    parents{
                      name
                      parents{
                        name
                        parents{
                          name
                          parents{
                            name
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
          score
          datatypeScores{
            id
            score
          }
        }
      }
    }
  }
"
  
  base_url <- "https://api.platform.opentargets.org/api/v4/graphql"
  variables <- list("ensemblId" = ensembl_id)
  post_body <- list(query = query_string, variables = variables)
  r <- POST(url=base_url, body=post_body, encode='json')
  
  data = httr::content(r)$data
  data3 = map_dfr(data$target$associatedDiseases$rows, function(x){
    c(setNames(c(x[[1]]$name, paste(unlist(x[[1]]$therapeuticAreas), collapse = " | "), paste(unlist(x[[1]]$parents), collapse = " | "), x[[2]]), 
               c("Disease_name", "Therapeutic_area", "Disease_parents", "Disease_score")), 
      x[[3]] %>% map_dfr(unlist) %>% pivot_wider(names_from = "id", values_from = "score"))
  }) %>% mutate(Kidney = factor(replace_na(str_extract(Disease_parents, "kidney disease|renal system measurement"), "Other"),
                                levels = c("kidney disease", "renal system measurement", "Other"), labels = c("Kidney disease", "Renal measurement", "Other"))) %>%
    group_by(Kidney) %>% slice_max(Disease_score, n = 25) %>% ungroup
  
  diseaseTypes = c("Disease_score", "genetic_association", "somatic_mutation", "known_drug", "affected_pathway", "literature", "rna_expression", "animal_model")
  diseaseTypeLabels = c("Overall\nassociation score", "Genetic associations", "Somatic mutations", "Drugs", "Pathways\n& systems biology",
                        "Text mining", "RNA expression", "Animal models")
  
  data4 = data3 %>%
    dplyr::mutate(across(.cols = !Disease_name & !Therapeutic_area & !Disease_parents & !Kidney, .fns = as.numeric)) %>%
    pivot_longer(cols = -Disease_name & -Therapeutic_area & -Disease_parents & -Kidney, names_to = "Scores", values_to = "Score") %>%
    mutate(Scores = factor(Scores, levels = diseaseTypes, labels = diseaseTypeLabels))
  
  areas = unique(as.vector(str_split(data3$Therapeutic_area, " \\| ", simplify = TRUE)))
  areas2 = map_dfc(as.list(areas[areas != ""]), function(x) setNames(data.frame(str_detect(data3$Therapeutic_area, x)), x)) %>%
    mutate("Disease_name" = data3$Disease_name) %>%
    pivot_longer(cols = -Disease_name) %>%
    `colnames<-`(c("Disease_name", "Area", "InArea"))

  data5 = data4 %>% left_join(areas2, by = "Disease_name", multiple = "all") %>%
    mutate(Disease_name = factor(Disease_name, levels = data3$Disease_name)) %>%
    complete(Scores, nesting(Disease_name, Therapeutic_area, Area, Kidney, InArea))

  plot1 = ggplot(data5, aes(x = Scores, y = Disease_name, fill = Score)) +
    geom_tile(col = "grey90") +
    scale_fill_viridis(na.value = "white") + scale_y_discrete(limits = rev) + scale_x_discrete(drop = FALSE) +
    facet_grid(rows = vars(Kidney), scales = "free", space = "free") +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), plot.title = element_text(hjust = 0.5), 
          strip.background = element_blank(), strip.text.y = element_blank(),
          text = element_text(size = 18)) + 
    labs(x = NULL, y = NULL, title = paste("Association scores", gene_id))
  
  plot2 = ggplot(data5, aes(x = Area, y = Disease_name, fill = InArea)) +
    geom_tile(col = "grey90") +
    scale_fill_manual(values = c("FALSE" = "white", "TRUE" = "black")) + scale_y_discrete(limits = rev) + 
    facet_grid(rows = vars(Kidney), scales = "free", space = "free") +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), plot.title = element_text(hjust = 0.5),
          strip.text.y = element_text(angle = 0), axis.text.y = element_blank(), axis.ticks.y = element_blank(),
          text = element_text(size = 18)) + 
    labs(x = NULL, y = NULL, title = "Disease areas") + guides(fill = "none")
  
  plots = plot1 + plot2 + plot_layout(guides = "collect", widths = c(1, 1.5))
  return(list("plots" = plots, "data4" = data4, "data5" = data5))
}

```

### Combined plot
```{r OpenTargets combined plot, fig.height=9, fig.width=12}
# gene.list = list("CR1", "C1QA", "C1QB", "C1QC", "C1R", "C1S", "C2", "C3", "C4A", "C4B", "C5")
gene.list = list("CCL17", "IL3RA", "LY96", "CXCR3", "PLG")

OT_list = map_dfr(gene.list, function(gene){
  data.frame("Gene" = gene, OT_query(gene)$data4)
})
data.OT = OT_list %>%
  filter(Kidney %in% c("Kidney disease", "Renal measurement")) %>%
  complete(Gene, Scores, nesting(Disease_name, Kidney))
dis.order = data.OT %>% filter(Scores == "Overall\nassociation score" & !is.na(Score)) %>%
                     arrange(Gene, -Score) %>% select(Disease_name) %>% unique
data.OT = data.OT %>% mutate(Disease_name = factor(Disease_name, levels = dis.order$Disease_name))
ggplot(data.OT, aes(x = Gene, y = Disease_name, fill = Score)) +
  geom_tile(col = "grey90") +
  scale_fill_viridis(na.value = "white") + scale_y_discrete(limits = rev) + scale_x_discrete(drop = FALSE) +
  facet_grid(Kidney ~ Scores, scales = "free", space = "free") +
  theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1), strip.text.y = element_text(angle = 0))

ggsave("Output/OTquery_CSLassets-Combined_TW.pdf", height = 9, width = 12)

```

### Single plots
```{r OpenTargets single plots, fig.height=9, fig.width=18}
gene.list = list("CCL17", "IL3RA", "LY96", "CXCR3", "PLG")

plot.list = map(gene.list, function(gene){
  data.OT = OT_query(gene)$data4 %>%
    filter(Kidney %in% c("Kidney disease", "Renal measurement")) %>%# & !is.na(Score)) #& Scores == "genetic_association" & 
    complete(Scores, nesting(Disease_name, Kidney))
  dis.order = data.OT %>% filter(Scores == "Overall\nassociation score" & !is.na(Score)) %>%
    arrange(Score) %>% select(Disease_name) %>% unique
  data.OT = data.OT %>% 
    mutate(Disease_name = factor(Disease_name, levels = dis.order$Disease_name),
           Overall = factor(Scores == "Overall\nassociation score", levels = c(TRUE, FALSE)))
  ggplot(data.OT, aes(x = Scores, y = Disease_name, fill = Score)) +
    geom_tile(col = "grey90") +
    # scale_x_discrete(drop = FALSE) +
    facet_grid(Kidney ~ Overall, scales = "free", space = "free") +
    labs(x = NULL, y = NULL, title = gene) +
    theme_bw() + 
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1), 
          strip.text.y = element_text(angle = 0), plot.title = element_text(hjust = 0.5),
          strip.background.x = element_blank(), strip.text.x = element_blank())
})
max.score = max(unlist(map(plot.list, function(plot) plot$data$Score)), na.rm = TRUE)
plot.list = map(plot.list, function(plot) plot + scale_fill_viridis(na.value = "white", limits = c(0, max.score)))
wrap_plots(plot.list, guides = "collect", heights = c(1,2))

ggsave("Output/OTquery_CSLassets-Singles_TW.pdf", height = 9, width = 18)

```


## AKI

### Urine CSA-AKI
Fetuin-A = AHSG = P02765 = FETUA_Human = Alpha-2-HS-glycoprotein

- Paper URL: https://www.frontiersin.org/articles/10.3389/fbioe.2022.1002853/full
- Data downloaded from: https://www.frontiersin.org/articles/10.3389/fbioe.2022.1002853/full#supplementary-material
  - Table 4.xls
- Gene names downloaded from: https://www.genenames.org/download/custom/
  - Selected columns: all standards + "Alias names", "NCBI Gene ID", "Ensembl gene ID", "UniProt ID"
  - Saved as .txt

```{r Fetuin-A, fig.height=4, fig.width=7}
raw.data = fread("Input/Bai2022_UrinaryProteomics_Table 4.csv")
HGNC = data.table::fread("../Other/Input/HGNC_GeneNames_17-05-23.txt", quote = "")

data = raw.data %>%
  select(-ProteinGroup) %>%
  mutate(Protein = HGNC$`Approved symbol`[match(str_extract(Protein, "(?<=sp\\|).*(?=\\|)"),
                                                 HGNC$`UniProt ID(supplied by UniProt)`)]) %>%
  filter(!is.na(Protein)) %>% # 16/1942 Proteins not matched, 1926 remaining
  column_to_rownames("Protein") %>%
  t %>% as.data.frame %>%
  mutate(Patient = factor(rep(1:6, 2)),
         TimePoint = factor(rep(c("After surgery (Day 1)", "Before surgery"), each = 6),
                            levels = c("Before surgery", "After surgery (Day 1)")))

data = data %>% mutate(norm_FetuinA = AHSG - CST3)

box.plot = ggplot(data, aes(x = TimePoint, y = PLAU, fill = TimePoint)) +
  geom_boxplot() +
  stat_compare_means(comparisons = list(c(1,2)), size = 3) +
  theme_bw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + xlab(NULL)

line.plot = ggplot(data, aes(x = TimePoint, y = PLAU, group = Patient, col = Patient)) +
  geom_line() + geom_point() +
  theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))

box.plot + line.plot
ggsave("Output/Bai2022_UrinaryProteomics_FetuinA.pdf", height = 4, width = 7)

```

### Mouse model of septic AKI

- Paper: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7710145/
- Data downloaded: Supplementary Table S1 ("162338_1_supp_596735_qgmqcy.xlsx")

```{r Septic-AKI mouse, fig.width=4, fig.height=4}
raw.data = read.xlsx("Input/AKI/162338_1_supp_596735_qgmqcy.xlsx", sheetIndex = 1, startRow = 2)
data = raw.data %>% dplyr::select(Gene.Name, contains("LFQ")) %>%
  column_to_rownames("Gene.Name") %>% t %>% as.data.frame %>%
  # dplyr::mutate(across(everything(), ~log2(.x+1))) %>%
  mutate(Animal = as.numeric(str_extract(rownames(.), "(?<=G)[:digit:]{1,2}"))) %>%
  arrange(Animal) %>%
  mutate(Group = rep(c("Day 0", "Day 2", "Day 7"), each = 6))

gene = "Ahsg"
ggplot(data, aes(x = Group, y = .data[[gene]], col = Group, fill = Group)) +
  geom_boxplot(alpha = 0.25, outlier.alpha = 0) + geom_jitter(width = 0.2, alpha = 0.2) +
  stat_compare_means(comparisons = list(c(1,2), c(1,3), c(2,3)), size = 4) +
  labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), 
       title = "Kidney proteomics\nSepsis-induced Kidney Injury mouse model\nLin | Mol Cell Proteomics 2020") +
  theme_bw() + 
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.title = element_text(hjust = 0.5, size = 12))

c("Ccl17", "Ly96", "Plg", "Il3ra", "Cxcr3")[c("Ccl17", "Ly96", "Plg", "Il3ra", "Cxcr3") %in% colnames(data)]

```


## Lupus Nephritis
Highlighted drugs:

- CSL305 - anti-**C2** mAb - dual FcRn blocker & complement CP & LP inhibitor
- CSL040 - soluble **CR1** - complement pathway inhibitor
- CSL300 - anti-**IL-6** - anti-inflammatory
- Vamifeport - Ferroportin/**SLC40A1**

Strategy taken: scRNAseq data of LN patients

Data downloaded from: https://singlecell.broadinstitute.org/single_cell/study/SCP279/amp-phase-1

- Paper 1: AMP Phase 1 - Dataset 2: Lupus Nephritis (Broad Institute) 
  - https://www.nature.com/articles/s41590-019-0398-x
  - Arazi *et al.*, The immune cell landscape in kidneys of patients with lupus nephritis, *Nature Immunology* 2019
- Paper 2: AMP Phase 1 - Dataset 3: Lupus Nephritis (NYU Langone - Metro)
  - https://www.nature.com/articles/s41590-019-0386-1
  - Der *et al.*, Tubular cell and keratinocyte single-cell transcriptomics applied to lupus nephritis reveal type I IFN and fibrosis relevant pathways, **Nature Immunology**, 2019

### Load data
```{r Load data}
annot = fread("Input/LupusNephritis/meta.gz")
data1 = fread("Input/LupusNephritis/exprMatrixSleBroad.tsv.gz")
data2 = fread("Input/LupusNephritis/exprMatrixSleMetro.tsv.gz")


```

### Set up query functions
```{r Query functions}

```

### Lupus bulk transcriptomics - longitudinal

- Paper: Banchereau et al., "Personalized Immunomonitoring Uncovers Molecular Networks that Stratify Lupus Patients", Cell 2016
- Link paper: https://www.sciencedirect.com/science/article/pii/S0092867416302641
```{r Lupus bulk transcriptomics, fig.width=4.5, fig.height=5}
data3 = getGEO("GSE65391")[[1]]
annot.pat = data3@phenoData@data %>%
  filter(`set:ch1` != "Technical_Replicate") %>%
  `colnames<-`(str_replace(str_remove(colnames(.), "\\:ch1"), " ", "_")) %>%
  dplyr::select(geo_accession, subject, disease_state, visit, disease_activity, sledai, treatment_lmm1, sledaic_lmm2)
annot.gene = data3@featureData@data
expr = data3@assayData$exprs %>% as.data.frame %>%
  dplyr::mutate(Gene = annot.gene$ILMN_Gene[match(rownames(.), annot.gene$ID)]) %>%
  group_by(Gene) %>% dplyr::summarise(across(everything(), max))
expr2 = expr %>%
  column_to_rownames("Gene") %>% t %>% as.data.frame %>%
  rownames_to_column("geo_accession") %>% inner_join(annot.pat, by = "geo_accession") %>%
  mutate(disease_activity = factor(ifelse(disease_state == "Healthy", "Healthy", paste0("SLE_", disease_activity)),
                                   levels = c("Healthy", paste0("SLE_", 1:3))))

pat.1 = expr2 %>% group_by(disease_activity) %>% dplyr::count() %>%
  mutate(Labels = paste0(disease_activity, " (", n, ")"))
pat.2 = expr2 %>% group_by(sledaic_lmm2) %>% dplyr::count() %>%
  mutate(Labels = paste0(sledaic_lmm2, " (", n, ")"))
expr2 = expr2 %>% mutate(disease_activity = factor(disease_activity, 
                                                   levels = pat.1$disease_activity, labels = pat.1$Labels),
                         sledaic_lmm2 = factor(sledaic_lmm2,
                                               levels = rev(pat.2$sledaic_lmm2), labels = rev(pat.2$Labels)))

gene.list = list("CCL17", "IL3RA", "PLG", "LY96", "CXCR3")#, "IFIT3", "CXCL9", "CXCL10", "CXCL11")
gene.list = list("LY96")#, "CXCR3")#, "CXCL10", "IFIT3")
gene.list = list("IL6R", "IL6", "IL5", "IL4")
plot.list = map2(gene.list, "disease_activity", function(gene, score){
  grouping = ifelse(score == "disease_activity", "Disease activity", "SLEDAI categories")
  comp.list = if(score == "disease_activity") list(c(1,2), c(2,3), c(3,4), c(1,3), c(1,4)) else list(c(1,2), c(2,3), c(3,4), c(4,5), c(5,6), c(3,5), c(2,5))
  ggplot(expr2, aes(x = .data[[score]], y = .data[[gene]], fill = .data[[score]], col = .data[[score]])) +
    geom_boxplot(alpha = 0.25, outlier.alpha = 0, col = "black") + geom_jitter(width = 0.2, alpha = 0.15) +
    stat_compare_means(comparisons = comp.list, na.rm = TRUE, size = 3, step.increase = 0.075) +
    labs(x = NULL, y = paste0(gene, "\nmRNA levels (log2)"), fill = grouping, col = grouping) +
    theme_bw() + 
    theme(axis.text.x = element_blank(), axis.ticks.x = element_blank())
})
wrap_plots(plot.list, guides = "collect") + 
  plot_annotation("SLE blood transcriptomics\nBanchereau Cell 2016") & theme(plot.title = element_text(hjust = 0.5))
ggsave("Output/SLE_BloodTranscriptomics_LY96_DA.pdf", width = 5, height = 5)

```

## CKD

### CKD plasma proteomics

- Paper: Medina et al., Proteomic signature associated with chronic kidney disease (CKD) progression identified by data-independent acquisition mass spectrometry, Clinical Proteomics 2023
- Data downloaded from: https://clinicalproteomicsjournal.biomedcentral.com/articles/10.1186/s12014-023-09405-0
  - Additional file 2

```{r CKD plasma proteomics, fig.width=4, fig.height=3}
data = read.csv("Input/12014_2023_9405_MOESM2_ESM.csv")
pat.counts = data %>% group_by(Progressor.type) %>% dplyr::count() %>% 
  ungroup %>% dplyr::select(n) %>% unlist
data = data %>%
  mutate(Progressor.type = factor(Progressor.type, levels = c("Stable", "Rapid"),
                                  labels = paste0(c("Stable", "Rapid"), " (", rev(pat.counts), ")")))

colnames(data)[-c(1,ncol(data))] = HGNC$`Approved symbol`[match(colnames(data)[-c(1,ncol(data))],
                                                                HGNC$`UniProt ID(supplied by UniProt)`)]
  
ggplot(data, aes(x = Progressor.type, y = CCT4, col = Progressor.type, fill = Progressor.type)) +
  geom_boxplot(alpha = 0.25, outlier.alpha = 0) + geom_jitter(alpha = 0.2, width = 0.2) +
  stat_compare_means(comparisons = list(c(1,2))) +
  labs(x = NULL, y = "PLG - Plasmin(ogen)\nProtein levels (log2)", title = "CKD patient plasma proteomics") +
  theme_bw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(), plot.title = element_text(hjust = 0.5))
ggsave("Output/CKD-plasma-proteomics_PLG.pdf", width = 4, height = 3)


```

## Other

### IL-6
```{r IL-6}

```